source/mqtt_agent.c

```diff
@@ -293,6 +307,8 @@ static bool addAwaitingOperation( MQTTAgentContext_t * pAgentContext,
     bool ackAdded = false;
     AckInfo_t * pendingAcks = pAgentContext->pPendingAcks;
 
+    assert( packetId != MQTT_PACKET_ID_INVALID );
```

Comment:
Assert check missing for the context also?
```suggestion
    assert( pAgentContext != NULL );
    assert( packetId != MQTT_PACKET_ID_INVALID );
```